### PR TITLE
Refactor chromosome file reading with pandas

### PR DESCRIPTION
**Pull request by IvoTod**

### CODE REVIEW

**Comment by IvoTod** [comment]
[SP23: Read chromosome file in chromosome view with pandas](https://trello.com/c/9gVIVJvO/1262-sp23-read-chromosome-file-in-chromosome-view-with-pandas)

File: wdae/wdae/chromosome/tests/fixtures/permissionDeniedPrompt.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This is a real permission denied prompt. The view has successfully sent the prompt.
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/permissionDeniedPrompt.md]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/data/inheritance_trio.ped [hunk 2]
```diff
@@ -0,0 +1,4 @@
+familyId	personId	dadId	momId	sex	status	role
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/data/inheritance_trio.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/data/inheritance_trio.vcf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+##fileformat=VCFv4.2
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/data/inheritance_trio.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/inheritance_trio.conf [hunk 4]
```diff
@@ -0,0 +1,75 @@
+[study]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/inheritance_trio/inheritance_trio.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.ped [hunk 5]
```diff
@@ -0,0 +1,5 @@
+familyId	personId	dadId	momId	sex	status	role
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.vcf [hunk 6]
```diff
@@ -0,0 +1,8 @@
+##fileformat=VCFv4.2
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/quads_in_child.conf [hunk 7]
```diff
@@ -0,0 +1,69 @@
+[study]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/quads_in_child.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.ped [hunk 8]
```diff
@@ -0,0 +1,5 @@
+familyId	personId	dadId	momId	sex	status	role
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.vcf [hunk 9]
```diff
@@ -0,0 +1,8 @@
+##fileformat=VCFv4.2
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/quads_in_parent.conf [hunk 10]
```diff
@@ -0,0 +1,66 @@
+[study]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/quads_in_parent.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.ped [hunk 11]
```diff
@@ -0,0 +1,9 @@
+familyId	personId	dadId	momId	sex	status	role
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.vcf [hunk 12]
```diff
@@ -0,0 +1,7 @@
+##fileformat=VCFv4.2
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/quads_two_families.conf [hunk 13]
```diff
@@ -0,0 +1,32 @@
+[study]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/quads_two_families.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.ped [hunk 14]
```diff
@@ -0,0 +1,5 @@
+familyId	personId	dadId	momId	sex	status	role
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.vcf [hunk 15]
```diff
@@ -0,0 +1,7 @@
+##fileformat=VCFv4.2
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/quads_variant_types.conf [hunk 16]
```diff
@@ -0,0 +1,29 @@
+[study]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/quads_variant_types.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/DAE.conf [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+[genomicScoresDB]
+confFile =%(wd)s/genomicScores.conf
+
+[genomesDB]
+confFile =%(wd)s/genomesDB.conf
+
+[defaultConfiguration]
+confFile = %(wd)s/defaultConfiguration.conf
+
+[gpfjs]
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/DAE.conf]: You can delete this section.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.ped (hunk 5 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.vcf (hunk 6 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/data/quads_in_child.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/quads_in_child.conf (hunk 7 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_child/quads_in_child.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.ped (hunk 8 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.vcf (hunk 9 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/data/quads_in_parent.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/quads_in_parent.conf (hunk 10 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_in_parent/quads_in_parent.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.ped (hunk 11 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.vcf (hunk 12 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/data/quads_two_families.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/quads_two_families.conf (hunk 13 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_two_families/quads_two_families.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.ped (hunk 14 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.ped]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.vcf (hunk 15 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/data/quads_variant_types.vcf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/quads_variant_types.conf (hunk 16 above)
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/studies/quads_variant_types/quads_variant_types.conf]: You can remove this file.

File: wdae/wdae/chromosome/tests/fixtures/hgTables.txt [hunk 18]
```diff
@@ -0,0 +1,30 @@
+#chrom	chromStart	chromEnd	name	gieStain
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/hgTables.txt]: Can you move this file to %(wd)/geneInfo/hgTables.txt

File: wdae/wdae/chromosome/views.py [hunk 19]
```diff
@@ -18,16 +18,22 @@ def __init__(self):
...
-                'end': int(row['chromEnd']),
-                'name': row['name'],
-                'gieStain': row['gieStain']
-            })
+        cols = ['chromStart', 'chromEnd', 'name', 'gieStain']
+        reader = reader.groupby('#chrom')[cols] \
+                       .apply(lambda x: x.values.tolist()) \
+                       .to_dict()
+
+        for k, v in reader.items():
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/views.py]: Try to make it with pandas.

File: wdae/wdae/chromosome/tests/fixtures/chr_test_data.yml [hunk 20]
```diff
@@ -0,0 +1,122 @@
+- bands:
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/fixtures/chr_test_data.yml]: Remove this file.

File: wdae/wdae/chromosome/tests/test_chromosome_api.py [hunk 21]
```diff
@@ -0,0 +1,10 @@
+import pytest
+
+pytestmark = pytest.mark.usefixtures('mock_gpf_instance')
+
+def test_chromosome_api_get(user_client, chromosome_test_data):
+    response = user_client.get('/api/v3/chromosomes')
+
+    assert response
+    assert response.status_code == 200
+    assert response.data == chromosome_test_data
```
Comment (nikolaystanishev) [inline wdae/wdae/chromosome/tests/test_chromosome_api.py]: Remove this and add a few small tests.

File: wdae/wdae/chromosome/views.py [hunk 22]
```diff
@@ -17,17 +17,17 @@ def __init__(self):
         reader = pd.read_csv(csvfile, delimiter='\t')
```
Comment (lchorbadjiev) [inline wdae/wdae/chromosome/views.py]: this is already parsed data frame so it probably better to rename it something 'chrom_df' or similar

**Pull request merged**